src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,68 @@
+package duke.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.DukeException;
+import duke.task.Task;
+
+/**
+ * Represents a class for managing data storage and retrieval.
+ */
+public class Storage {
+    private static final String LINE = "___________________________________\n";
+    private final String filepath;
+    /**
+     * Constructs a Storage object with the specified file path.
+     *
+     * @param filepath The path to the file where data will be stored.
+     */
+    public Storage(String filepath) {
+        this.filepath = filepath;
+    }
+
+    /**
+     * Loads tasks from the file and returns them as an ArrayList.
+     *
+     * @return An ArrayList containing the loaded tasks.
+     * @throws DukeException If there is an issue loading tasks from the file.
+     */
+    public ArrayList<Task> load() throws DukeException {
```

Comment:
perhaps this can be renamed to loadFromFile() to be more consistent with your other naming of methods like Task.readListFromFile